### PR TITLE
Add Ubuntu 20.04 cicd dockerfiles/buildscripts-develop

### PR DESCRIPTION
**Pull request by ndcgundlach**
## Change Description

Add docker files and build script entries for Ubuntu 20.04.

## Change Type
**Select ONE**
- [ ] Documentation

- [ ] Stability bug fix

- [ ] Other

- [ ] Other - special case

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

**Comment by kj4ezj** [comment]
I kicked off eosio-build-scripts [build 558](https://buildkite.com/EOSIO/eosio-build-scripts/builds/558) to test those YAML changes.

File: .cicd/platforms/pinned/ubuntu-20.04-pinned.dockerfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ENV VERSION 1
+# install dependencies.
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y git make \
+    bzip2 automake libbz2-dev libssl-dev doxygen graphviz libgmp3-dev \
+    autotools-dev python2.7 python2.7-dev python3 \
+    python3-dev python-configparser \
+    autoconf libtool g++ gcc curl zlib1g-dev sudo ruby libusb-1.0-0-dev \
+    libcurl4-gnutls-dev pkg-config patch vim-common jq
```
[the remark this answers is not in the document]
Comment (ndcgundlach) [reply .cicd/platforms/pinned/ubuntu-20.04-pinned.dockerfile]: done

File: .cicd/platforms/pinned/ubuntu-20.04-pinned.dockerfile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    rm -rf /llvm
+# build boost
+RUN curl -LO https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.bz2 && \
+    tar -xjf boost_1_72_0.tar.bz2 && \
+    cd boost_1_72_0 && \
+    ./bootstrap.sh --with-toolset=clang --prefix=/usr/local && \
+    ./b2 toolset=clang cxxflags='-stdlib=libc++ -D__STRICT_ANSI__ -nostdinc++ -I/usr/local/include/c++/v1 -D_FORTIFY_SOURCE=2 -fstack-protector-strong -fpie' linkflags='-stdlib=libc++ -pie' link=static threading=multi --with-iostreams --with-date_time --with-filesystem --with-system --with-program_options --with-chrono --with-test -q -j$(nproc) install && \
+    cd / && \
+    rm -rf boost_1_72_0.tar.bz2 /boost_1_72_0
+# install nvm
```
[the remark this answers is not in the document]
Comment (ndcgundlach) [reply .cicd/platforms/pinned/ubuntu-20.04-pinned.dockerfile:50]: done

File: .cicd/build-scripts.yml [hunk 3]
```diff
@@ -189,4 +202,20 @@ steps:
...
       - "apt update && apt upgrade -y && apt install -y git"
       - "./scripts/eosio_build.sh -y"
-    timeout: 180
\ No newline at end of file
+    timeout: 180
+
+  - label: ":ubuntu: Ubuntu 20.04 - Build UnPinned"
+    env:
+      DEBIAN_FRONTEND: "noninteractive" 
+      TZ: "America/New York"
```
Comment (kj4ezj) [inline .cicd/build-scripts.yml:210]: This `TZ` variable is missing from the pinned build, but the symbolic link is created in both. Does this variable need to be added there, too? Does the symbolic link make it unnecessary, and can it be removed here?

File: .cicd/platforms/pinned/ubuntu-20.04-pinned.dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (kj4ezj) [reply .cicd/platforms/pinned/ubuntu-20.04-pinned.dockerfile:50]: Please don't forget to remove the `nvm` code:
```dockerfile
# install nvm
RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.0/install.sh | bash
# load nvm in non-interactive shells
RUN cp ~/.bashrc ~/.bashrc.bak && \
    cat ~/.bashrc.bak | tail -3 > ~/.bashrc && \
    cat ~/.bashrc.bak | head -n '-3' >> ~/.bashrc && \
    rm ~/.bashrc.bak
```
It is dead code, we aren't using it anymore and our installation is janky.

File: .cicd/platforms/unpinned/ubuntu-20.04-unpinned.dockerfile [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    ./b2 --with-iostreams --with-date_time --with-filesystem --with-system --with-program_options --with-chrono --with-test -j$(nproc) install && \
+    cd / && \
+    rm -rf boost_1_71_0.tar.bz2 /boost_1_71_0
+# install nvm
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.0/install.sh | bash
+# load nvm in non-interactive shells
+RUN cp ~/.bashrc ~/.bashrc.bak && \
+    cat ~/.bashrc.bak | tail -3 > ~/.bashrc && \
+    cat ~/.bashrc.bak | head -n '-3' >> ~/.bashrc && \
+    rm ~/.bashrc.bak
```
Comment (kj4ezj) [inline .cicd/platforms/unpinned/ubuntu-20.04-unpinned.dockerfile:36]: Please remove the `nvm` code, as we are no longer using it.

**Pull request merged**